tests/collections/collection.test.ts

```diff
@@ -596,13 +1761,79 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
         },
         {
           "returnDocument": "after"
-        }          
+        }
       );
       assert.equal(findOneAndUpdateResp.ok, 1);
       assert.equal(findOneAndUpdateResp.value!._id, docId);
       assert.equal(findOneAndUpdateResp.value!.username, "aaronm");
       assert.equal(findOneAndUpdateResp.value!.address.city, undefined);
-    });      
+    });
+    it('should findOneAndUpdate with returnDocument before', async () => {
+      const docToInsert = createSampleDocWithMultiLevel();
+      const res = await collection.insertOne(docToInsert);
+      const docId = res.insertedId;
+      const cityBefore = docToInsert.address?.city;
+      const usernameBefore = docToInsert.username;
+      const findOneAndUpdateResp = await collection.findOneAndUpdate({ "_id": docId },
+        {
+          "$set": {
+            "username": "aaronm"
+          },
+          "$unset": {
+            "address.city": ""
+          }
+        },
+        {
+          "returnDocument": "before"
+        }
+      );
+      assert.equal(findOneAndUpdateResp.ok, 1);
+      assert.equal(findOneAndUpdateResp.value!._id, docId);
+      assert.equal(findOneAndUpdateResp.value!.username, usernameBefore);
+      assert.equal(findOneAndUpdateResp.value!.address.city, cityBefore);
+    });
+    it('should findOneAndUpdate with upsert true', async () => {
+      const res = await collection.insertOne(createSampleDocWithMultiLevel());
+      const newDocId = "123";
+      const findOneAndUpdateResp = await collection.findOneAndUpdate({ "_id": newDocId },
+        {
+          "$set": {
+            "username": "aaronm"
+          },
+          "$unset": {
+            "address.city": ""
+          }
+        },
+        {
+          "returnDocument": "after",
+          "upsert": true
+        }
+      );
+      assert.equal(findOneAndUpdateResp.ok, 1);
+      assert.equal(findOneAndUpdateResp.value!._id, newDocId);
+      assert.equal(findOneAndUpdateResp.value!.username, "aaronm");
+      assert.equal(findOneAndUpdateResp.value!.address, undefined);
+    });
+    it('should findOneAndUpdate with upsert true and returnDocument before', async () => {
+      const res = await collection.insertOne(createSampleDocWithMultiLevel());
+      const newDocId = "123";
+      const findOneAndUpdateResp = await collection.findOneAndUpdate({ "_id": newDocId },
+        {
+          "$set": {
+            "username": "aaronm"
+          },
+          "$unset": {
+            "address.city": ""
+          }
+        },
+        {
+          "returnDocument": "before",
+          "upsert": true
+        }
+      );
+      assert.equal(findOneAndUpdateResp.ok, 1);
+      assert.equal(findOneAndUpdateResp.value, undefined);
```

Comment:
Minor nitpick: I think returning `null` is more idiomatic if there's no result. And better to use `assert.strictEqual()` when working with `null` or `undefined` because of JavaScript's infamously strange type coercions.

Reply:
Fixed [9a79f8a](https://github.com/stargate/stargate-mongoose/pull/91/commits/9a79f8a6fe744f7319cdee1b34c4b71f3bc4f1e9)